map/src/context/RoutingManager.js

```diff
@@ -102,60 +103,439 @@ function filterMode(data) {
+            mergeStateObject(ctx.routeProviders, ctx.setRouteProviders, { paused: state });
+        },
+
+        CHOOSE(ctx = null, { type = null, router = null, profile = null } = {}) {
+            /*
+                Switch to selected "router" and/or "profile":
+
+                1. use current router|profile when (any)===null
+                2. use first available when router|profile is not applicable
+                3. type is changed automatically according to applicated router
+
+                Examples:
+
+                CHOOSE(ctx, { profile }) - switch profile, keep router
+                CHOOSE(ctx, { router }) - switch router, try to keep profile
+                CHOOSE(ctx, { router, profile }) - switch both (if applicable)
+
+                Special case (used for searchParams / share route link):
+
+                CHOOSE(ctx, { type, [profile] }) - select 1st type's provider
+
+                Return: updated routeProviders (use if you need) or null
+            */
+            const routeProviders = ctx?.routeProviders;
+
+            if (!ctx || !routeProviders) throw(new Error('invalid route.CHOOSE() call'));
+
+            if (type && !router) {
+                router = routeProviders.getProviderByType(type)?.key ?? 'osmand';
+                // console.log('type', type, 'router', router);
+            }
+
+            // use current values if not defined
+            router = router ?? routeProviders.router;
+            profile = profile ?? routeProviders.profile;
+
+            const providers = this.allProviders();
+
+            if (!providers.find(r => r.key === router)) {
+                router = providers[0].key ?? 'osmand'; // fallback
+            }
+
+            const profiles = this.allProfiles(router);
+
+            if (!profiles.find(p => p.key === profile)) {
+                profile = profiles[0].key ?? 'car'; // fallback
+            }
+
+            if (router && profile) {
+                type = this.getProvider(router)?.type ?? 'osmand'; // fallback
+                // console.log('choosen:', type, router, profile);
+                return mergeStateObject(routeProviders, ctx.setRouteProviders, { type, router, profile });
+            }
+
+            return routeProviders; // unmodified
+        },
+
+        /*
+            providersOSRM and providersOsmAnd are loaded by loadRouterProviders()
+            Nevertheless, providersOsmAnd might be used before they are loaded
+            Therefore, OsmAnd must be initialized with minimal-fallback data
+        */
+        providersOSRM: [],
+        providersOsmAnd: [Object.assign({}, preloaded,
+            { key: preloaded.router, profiles: [{ key: 'car', name: 'Car' }] })],
+
     };
 }
 
 // load and validate OSRM and OsmAnd routing providers
 async function loadRouteProviders({ routeProviders, setRouteProviders, creatingRouteMode = null, setCreatingRouteMode = null }) {
+    const nameAliases = {
```

Comment:
use this https://github.com/osmandapp/web/commit/b78f3ae8275550adc32212753f91e02915b121c3